doc/lua_api.txt

```diff
@@ -4868,6 +4868,9 @@ Utilities
           get_sky_as_table = true,
           -- VoxelManip:get_light_data accepts an optional buffer argument (5.7.0)
           get_light_data_buffer = true,
+          -- When using an appropriate backend (not the deprecated "files" backend),
+          -- mod storage data is not necessarily all kept in RAM (5.7.0)
+          mod_storage_on_disk = true,
```

Comment:
fuzzy as it is what is the practical application?

Reply:
It means that, when using a non-deprecated backend, the size of mod storage is not limited by the available RAM. This doesn't include the `dummy` backend, which explicitly keeps everything in RAM. Perhaps I should change this comment.

Reply:
The documentation has been changed.

Reply:
No I understand what it means, but what are mods expected to do with this information?

Reply:
If they detect this feature, they can basically store as much data as they want in mod storage.

Reply:
They wouldn't have to insecurely require lsqlite unless they needed a full-on relational database.

Reply:
The problem I see here is that the feature entry only says "_maybe_ it won't all be in RAM". If a mod relies on this and stores lots of data then -- depending on circumstances it cannot determine -- the engine might still run out of RAM.
For a mod the only safe way is to not store much data regardless of this feature flag. Therefore it's not useful.

Reply:
The updated documentation should be more clear about this.

Reply:
![grafik](https://user-images.githubusercontent.com/1042418/189991849-ba63f408-8bf6-46a7-8213-98133edab8c5.png)
Maybe this diagram makes it clearer.
The problem I see is that mods cannot know _for sure_ (\*) that storing lots of data is safe.
(\*) It is certainly possible to open world.mt and read the current backend. Aside from being obviously unportable we shouldn't encourage or support this way (API guarantees).

This is fundamentally the same reason why #12167 was rejected.

Reply:
It's unfortunate, but mods can specify in their READMEs that a backend should be used other than "files" or "dummy".